corehq/apps/domain/decorators.py

```diff
@@ -675,3 +675,21 @@ def _wrapped(*args, **kwargs):
         return _wrapped
 
     return _dec
+
+
+def allow_cors(allowed_methods):
+    def decorator(view_func):
+        @wraps(view_func)
+        def wrapped_view(request, *args, **kwargs):
+            if request.method == "OPTIONS":
+                response = HttpResponse(allowed_methods)
```

Comment:
Should this be `JsonResponse`? Didn't know it was valid to pass a list as response `content`.

Reply:
This does work, though wasn't intentional ([here's why if you're curious](https://github.com/django/django/blob/main/django/http/response.py#L353-L357) - maybe related to chunked encoding). Anyway, I read up and the response body of an `OPTIONS` request is undefined, so I just removed it.

The API version returned a comma separated list of the methods, which I left untouched.

```
$ curl -X OPTIONS https://www.commcarehq.org/a/sabbatical/api/v0.5/user/ -i
HTTP/1.1 200 OK
Date: Tue, 06 Apr 2021 08:37:31 GMT
Content-Type: text/html; charset=utf-8
Content-Length: 8
Connection: keep-alive
Server: nginx
Access-Control-Allow-Origin: *
Access-Control-Allow-Headers: Content-Type, Authorization
Allow: GET,POST
Strict-Transport-Security: max-age=300; includeSubDomains
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
HTTP_X_OPENROSA_VERSION: 1.0
X-Frame-Options: DENY
Vary: Accept-Language, Cookie
Content-Language: en
Cache-Control: private, no-cache, no-store, must-revalidate, proxy-revalidate
Expires: Thu, 01 Dec 1994 16:00:00 GMT
Pragma: no-cache
Age=1209600; Path=/; SameSite=Lax; Secure
Strict-Transport-Security: max-age=300
X-XSS-Protection: 1; mode=block
X-Content-Type-Options: nosniff

GET,POST
```